astropy/nddata/ccddata.py

```diff
@@ -330,6 +328,11 @@ def to_hdu(self, hdu_mask='MASK', hdu_uncertainty='UNCERT',
             # Note that until astropy/astropy#3967 is closed, the extend
             # will fail if there are comment cards in the WCS header but
             # not header.
+            if not isinstance(self.wcs, WCS):
```

Comment:
This is just my personal opinion but we need to keep this working, otherwise ccdproc will break for many users (because trimming and other operations rely on slicing CCDData).